crates/teloxide/examples/error_handler.rs

```diff
@@ -0,0 +1,113 @@
+use std::sync::Arc;
+
+use teloxide::{
+    adaptors::throttle::Limits, dptree::di::DependencySupplier, prelude::*,
+    utils::command::BotCommands,
+};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+enum PublicError {
+    #[error("Hey, there's been a mistake!")]
+    Dummy,
+    #[error("Private error occured")]
+    Private(#[from] PrivateError),
+}
+
+#[derive(Debug, Error)]
+enum PrivateError {
+    #[error("Teloxide request error: {0}")]
+    Teloxide(#[from] teloxide::RequestError),
+}
+
+impl From<teloxide::RequestError> for PublicError {
+    fn from(value: teloxide::RequestError) -> Self {
+        Self::Private(PrivateError::Teloxide(value))
+    }
+}
+
+#[derive(Clone, BotCommands)]
+#[command(rename_rule = "snake_case")]
+enum Command {
+    Dummy,
+}
+
+type HandlerResult = Result<(), PublicError>;
+type Bot = teloxide::adaptors::Throttle<teloxide::Bot>;
+
+#[tokio::main]
+async fn main() {
+    pretty_env_logger::init();
+    log::info!("Starting error_handler example bot...");
+
+    let bot = teloxide::Bot::from_env().throttle(Limits::default());
+
+    let schema = dptree::entry().branch(
+        Update::filter_message()
+            .branch(
+                dptree::entry()
+                    .filter_command::<Command>()
+                    .branch(dptree::case![Command::Dummy].endpoint(invoke_dummy_error)),
+            )
+            .branch(dptree::endpoint(say_hello)),
+    );
+
+    Dispatcher::builder(bot, schema)
+        .error_handler(Arc::new(error_handler))
+        .build()
+        .dispatch()
+        .await;
+}
+
+async fn error_handler(error: PublicError, deps: Option<DependencyMap>) {
```

Comment:
the map can already be empty, no need for an `Option`

Reply:
Just wanted to highlight that in some cases there will be no meaningful data

Reply:
I don't think this is necessary. error handler may run at any basically point during update handling, so it can't assume much about available dependencies.